enterprise/views.py

```diff
@@ -375,32 +465,71 @@ def get(self, request):
+        if not (enterprise_customer_uuid and success_url and failure_url):
+            error_code = 'ENTGDS001'
+            LOGGER.error(
+                'Error: one or more of the following values was falsy: '
+                'enterprise_customer_uuid: {enterprise_customer_uuid}, '
+                'success_url: {success_url}, '
+                'failure_url: {failure_url}. '
+                'The following error code was reported to the user: {error_code}'.format(
+                    enterprise_customer_uuid=enterprise_customer_uuid,
+                    success_url=success_url,
+                    failure_url=failure_url,
+                    error_code=error_code,
+                )
+            )
+            messages.add_generic_error_message_with_code(request, error_code)
+            errored = True
 
-            consent_record = get_data_sharing_consent(
-                request.user.username,
+        try:
+            enterprise_customer, errored = self._resolve_enterprise_customer_for_get(
+                request,
+                course_id,
+                program_uuid,
                 enterprise_customer_uuid,
-                program_uuid=program_uuid,
-                course_id=course_id
+                errored
             )
-            if consent_record is None or not consent_record.consent_required():
-                raise Http404
-
-            enterprise_customer = consent_record.enterprise_customer
-        else:
-            if not request.user.is_staff:
-                raise PermissionDenied()
+        except NotConnectedToOpenEdX:
             enterprise_customer = get_enterprise_customer_or_404(enterprise_customer_uuid)
+            context_data = get_global_context(request, enterprise_customer)
+            # need to have the render here so we don't proceed any further in the view
+            return render(
+                request,
+                ENTERPRISE_GENERAL_ERROR_PAGE,
+                context=context_data,
+                status=404,
+            )
 
         context_data = get_global_context(request, enterprise_customer)
-        context_data.update(
-            self.get_course_or_program_context(enterprise_customer, course_id=course_id, program_uuid=program_uuid)
-        )
+        try:
+            updated_context_dict = self.get_course_or_program_context(
+                enterprise_customer,
+                course_id=course_id,
+                program_uuid=program_uuid
+            )
+            context_data.update(updated_context_dict)
+        except Http404:
+            error_code = 'ENTGDS000'
+            LOGGER.error(
+                'CourseCatalogApiServiceClient is improperly configured.'
+                'Returned error code {error_code} to user'.format(
+                    error_code=error_code,
+                )
+            )
+            messages.add_generic_error_message_with_code(request, error_code)
+            errored = True
+
+        if errored:
```

Comment:
i thought it was more concise to just render the error page with a few different error messages via 1 check later in the view if there were any errors... that technically does change the behavior of the view slightly though... i went with this solution because when using the helper methods that could choke in a few different ways, it was simpler just to keep tabs on whether or not anything went wrong throughout the process, and throw an error page at the end. questioning that now

Reply:
if the enterprise_customer_uuid is not valid here though, later on down the road to get the global context, things will choke (because getting the global context requires an enterprise customer). consider adding another render here, but with a more sparse context